### PR TITLE
LPS-96251

### PR DESCRIPTION
**Pull request by Alec-Shay**
Relevant tickets:

https://issues.liferay.com/browse/LPP-33892
https://issues.liferay.com/browse/LPS-96251

This is a fix for a user experience issue with the error message for a failed CAPTCHA attempt. If reCAPTCHA is configured for users, and a user fails to validate with it, the error message that appears references "Text validation." On the user end though, there is no text validation to be seen, so it may be confusing to users what the error means.

My proposal, to avoid having to make a big change, is to simply reword it to say "User verification failed" instead, since to a regular user's perspective, the CAPTCHA might have nothing to do with text, and it is the user being verified. An alternative would potentially be to also try to determine what type the CAPTCHA is configured to use and change the error message based on that (possibly creating one or more new Exception types for it), but I was not sure as to how to detect the type since for reCAPTCHA that information seems to not be configured locally, so I think this change will probably be simpler and still make sense for users.

Of course I think we would still be open to other suggestions for how to go about this, but let me know if there are any other thoughts on this. Thanks!

### CODE REVIEW

**Comment by Alec-Shay** [comment]
ci:test:relevant

**Comment by Alec-Shay** [comment]
ci:test:sf

**Comment by ChrisKian** [comment]
Hey @Alec-Shay,
I agree with your thinking about how a simple rephrase is much easier than an overhaul of error detection and handling.  That being said, I have two issues with this PR.

First, a SF error was thrown, and it looks like it may be related.

Second, you are changing the error message for the "CaptchaTextException", but it seems it should stay the same given the exception class name.  Can you look into creating a new exception class and throwing/catching that one as needed instead?  Or, if you can you definitively say for certain the "CaptchaTextException" class isn't used anywhere else, it may work better to use your original solution AND change the exception class name.

Let me know if you have any questions, thanks!

**Comment by Alec-Shay** [comment]
@ChrisKian Hmmm... Thanks for catching the SF error, I'll take another look at that.

That makes sense the exception class name probably makes sense to stay that way too. I guess it may make sense to make a new Exception class for it, though I'll take another look to see whether there's any places that would make more sense to use. I suspect we might not be able to change the Exception class name at this point (well, maybe in master we can, but I don't know for sure if we'd want to), but a new one probably shouldn't cause problems.
